### PR TITLE
[content-init] Propagate sensible error messages

### LINKED ISSUE #7605: [content-initializer] Provide actionable feedback when content init fails
## Is your feature request related to a problem? Please describe
Today content initializer failure is a frustrating experience. Not only does the workspace not start, but the error message isn't helpful either. 

## Describe the behaviour you'd like
We must aim to provide a message/feedback which gives more insight into what went wrong, for example:
- when git operations fail, show the output of Git,
- when prebuild operations fail highlight that that's the case.

## Describe alternatives you've considered
Something will always fail - there is no alternative.

### PR DESCRIPTION
**Pull request by csweichel**
## Description
This PR provides more detailed error messages when the content initializer fails. It does that by passing a unix pipe to the content initializer process started by ws-daemon, to which that process can write error messages. We can't just use the stderr/stdout of the content initializer process because that's too verbose and JSON formatted. With this design we have more control over the final error message.

## Related Issue(s)

Fixes #7605

## How to test
- Open repo which is known to fail, e.g. https://github.com/csweichel/gitpod-hello-ui-demo (until #7749 is merged)
- Start a workspace, stop it, delete its backup and try to start it again. You should see something like

## Release Notes

```release-note
Improved feedback when content initialisation fails
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **csweichel** after the PR has been reviewed.

Associated issue: *#7605*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/content-service/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service/OWNERS)**
- **[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by Furisto — APPROVED** [review]

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 4423a23c70a4f197262784f735d321ecfbb78266</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7753#pullrequestreview-859515036" title="Approved">Furisto</a>*

Associated issue: *#7605*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/content-service/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service/OWNERS)~~ [Furisto]
- ~~[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)~~ [Furisto]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
